framework/common/encryption.cpp

```diff
@@ -88,11 +88,7 @@ diffie_hellman::diffie_hellman() : m_dh(nullptr), m_pubkey(nullptr)
     BN_bn2bin(pub_key, m_pubkey);
 }
 
-diffie_hellman::~diffie_hellman()
-{
-    delete m_pubkey;
-    DH_free(m_dh);
-}
+diffie_hellman::~diffie_hellman() { DH_free(m_dh); }
```

Comment:
This can't be right... The thing that is used inside m_dh is a BigNum, while m_pubkey is a copy of that that is allocated with our `new[]` and that is filled up with BN_bn2bin(). So the delete[] really is needed IMO...